library/src/lib/calendar/calendar-views/calendar-day-view/calendar-day-view.component.ts

```diff
@@ -0,0 +1,521 @@
+                case('Enter'): {
+                    event.preventDefault();
+                    this.selectDate(cell);
+                    this.newFocusedDayId = cell.id;
+                    break;
+                }
+                case('ArrowUp'): {
+                    event.preventDefault();
+                    if (grid.y > 0) {
+                        this.newFocusedDayId = this.dayViewGrid[grid.y - 1][grid.x].id;
+                    } else {
+                        this.selectPreviousMonth();
+                        this.newFocusedDayId = this.dayViewGrid[this.dayViewGrid.length - 1][grid.x].id;
+                    }
+                    break;
+                }
+                case('ArrowDown'): {
+                    event.preventDefault();
+                    if (grid.y < this.dayViewGrid.length - 1) {
+                        this.newFocusedDayId = this.dayViewGrid[grid.y + 1][grid.x].id;
+                    } else {
+                        this.selectNextMonth();
+                        this.newFocusedDayId = this.dayViewGrid[0][grid.x].id;
+                    }
+                    break;
+                }
+                case('ArrowLeft'): {
+                    event.preventDefault();
+                    if (grid.x > 0) {
+                        this.newFocusedDayId = this.dayViewGrid[grid.y][grid.x - 1].id;
+                    } else if (grid.y > 0) {
+                        this.newFocusedDayId = this.dayViewGrid[grid.y - 1][this.dayViewGrid[0].length - 1].id;
+                    } else {
+                        this.selectPreviousMonth();
+                        this.newFocusedDayId =
+                            this.dayViewGrid[this.dayViewGrid.length - 1][this.dayViewGrid[0].length - 1].id
+                        ;
+                    }
+                    break;
+                }
+                case('ArrowRight'): {
+                    event.preventDefault();
+                    if (grid.x < this.dayViewGrid[0].length - 1) {
+                        this.newFocusedDayId = this.dayViewGrid[grid.y][grid.x + 1].id;
+                    } else if (grid.y < this.dayViewGrid.length - 1) {
+                        this.newFocusedDayId = this.dayViewGrid[grid.y + 1][0].id;
+                    } else {
+                        this.selectNextMonth();
+                        this.newFocusedDayId = this.dayViewGrid[0][0].id;
+                    }
+                    break;
+                }
+            }
+        }
+
+        if (this.newFocusedDayId) {
+            this.focusElement(this.newFocusedDayId);
+        }
+    }
+
+    /** @hidden */
+    public ngOnChanges(changes: SimpleChanges): void {
+        if (!changes) {
+            return;
+        }
+        // To Prevent from multiple refreshing of calendar grid.
+        const keys = Object.keys(changes);
+        const isAnyChanged: boolean = !!keys.find(key => changes[key].currentValue !== changes[key].previousValue);
```

Comment:
Not sure why this is necessary? Did it refresh every tick for some reason?

Reply:
I removed it.